packages/devtools_app/test/flutter/split_test.dart

```diff
@@ -0,0 +1,276 @@
+
+        // Drag past the right end of the widget.
+        await tester.drag(find.byKey(_split.dividerKey), const Offset(0, 450));
+        await tester.pumpAndSettle();
+        expect(tester.element(find.byWidget(_w1)).size, const Size(800, 590));
+        expect(tester.element(find.byWidget(_w2)).size, const Size(800, 0));
+
+        // Make sure we can't overdrag.
+        await tester.drag(find.byKey(_split.dividerKey), const Offset(0, 200));
+        await tester.pumpAndSettle();
+        expect(tester.element(find.byWidget(_w1)).size, const Size(800, 590));
+        expect(tester.element(find.byWidget(_w2)).size, const Size(800, 0));
+
+        // Drag back past the left end of the widget.
+        await tester.drag(find.byKey(_split.dividerKey), const Offset(0, -600));
+        await tester.pumpAndSettle();
+        expect(tester.element(find.byWidget(_w1)).size, const Size(800, 0));
+        expect(tester.element(find.byWidget(_w2)).size, const Size(800, 590));
+
+        // Make sure we can't overdrag.
+        await tester.drag(find.byKey(_split.dividerKey), const Offset(0, -200));
+        await tester.pumpAndSettle();
+        expect(tester.element(find.byWidget(_w1)).size, const Size(800, 0));
+        expect(tester.element(find.byWidget(_w2)).size, const Size(800, 590));
+      });
+    });
+
+    group('resizes contents', () {
+      testWidgets('in a horizontal layout', (WidgetTester tester) async {
+        buildSplitAndChildren(Axis.horizontal, initialFirstFraction: 0.0);
+        await tester.pumpWidget(wrap(
+          Center(
+            child: SizedBox(width: 300.0, height: 300.0, child: _split),
+          ),
+        ));
+        expect(tester.element(find.byWidget(_w1)).size, const Size(0, 300));
+        expect(tester.element(find.byWidget(_w2)).size, const Size(290, 300));
+
+        await tester.pumpWidget(wrap(
+          Center(
+            child: SizedBox(width: 200.0, height: 200.0, child: _split),
+          ),
+        ));
+        expect(tester.element(find.byWidget(_w1)).size, const Size(0, 200));
+        expect(tester.element(find.byWidget(_w2)).size, const Size(190, 200));
+      });
+
+      testWidgets('in a vertical layout', (WidgetTester tester) async {
+        buildSplitAndChildren(Axis.vertical, initialFirstFraction: 0.0);
+        await tester.pumpWidget(wrap(
+          Center(
+            child: SizedBox(width: 300.0, height: 300.0, child: _split),
+          ),
+        ));
+        expect(tester.element(find.byWidget(_w1)).size, const Size(300, 0));
+        expect(tester.element(find.byWidget(_w2)).size, const Size(300, 290));
+
+        await tester.pumpWidget(wrap(
+          Center(
+            child: SizedBox(width: 200.0, height: 200.0, child: _split),
+          ),
+        ));
+        expect(tester.element(find.byWidget(_w1)).size, const Size(200, 0));
+        expect(tester.element(find.byWidget(_w2)).size, const Size(200, 190));
+      });
+    });
+  });
+}
+
+Widget _w1, _w2;
```

Comment:
nit: instead of making _w1, _w2, and _split 
surprise side effects of
buildSplitAndChildren why not use keys and lookup the widgets bsed on keys instead of using 
find.byWidget?

Reply:
Good idea, I've switched to keys for the child widgets.  Because we want to do other logic with the split, like rebuilding it in different places, I've kept that one as a return value on this method.